uplink/src/collector/tcpjson.rs

```diff
@@ -70,29 +78,46 @@ impl Bridge {
                     }
                     action = self.actions_rx.recv_async() => {
                         let action = action?;
-                        error!("Bridge down!! Action ID = {}", action.action_id);
-                        let status = ActionResponse::failure(&action.action_id, "Bridge down");
-                        if let Err(e) = self.action_status.fill(status).await {
-                            error!("Failed to send busy status. Error = {:?}", e);
+                        let action_id = action.action_id.clone();
+                        if self.actions_tx.send(action).is_err() {
```

Comment:
There is an implicit assumption that only one client should respond to an action. This should be documented.

Reply:
So multiple apps can still send action responses to the same, we expect atleast one connected app to send action response or else we timeout, it doesn't matter which app sent a response as long as the action_id is the same as in current_action.

Reply:
Just tested it and the actions are delivered to any one client at random. This needs to be figured out.

Reply:
Timeouts can overwrite genuine action responses.

Reply:
If any client disconnects, it will send a bridge disconnected status that can overwrite genuine responses.

Reply:
> Timeouts can overwrite genuine action responses.


> If any client disconnects, it will send a bridge disconnected status that can overwrite genuine responses.

Yes, we will have to rework how we do current actions in order to support multi-app feature.